go/vt/vtgate/executor.go

```diff
@@ -1010,8 +1010,13 @@ func (e *Executor) getPlan(vcursor *vcursorImpl, sql string, comments sqlparser.
 	_, _ = planHash.Write(hack.StringBytes(query))
 	planKey := hex.EncodeToString(planHash.Sum(nil))
 
-	if plan, ok := e.plans.Get(planKey); ok {
-		return plan.(*engine.Plan), nil
+	if qo.cachePlan() && sqlparser.CachePlan(statement) {
```

Comment:
nit: reverse the order of the check as `qo.cachePlan()` will most likely will always be true, so it will only depend on `sqlparser.CachePlan(statement)`